File: solutions/258/main.py
class Solution:
    def addDigits(self, num: int) -> int:
        while num >= 10:
            total = 0 
            while num >= 1:
                total += num % 10 
                num //= 10 
            num = total  
        return num 

if __name__ == "__main__": 
    s = Solution()

    # num = 38
    # assert s.addDigits(num) == 2 

    num = 10 
    assert s.addDigits(num) == 1

File: solutions/258/main.rs
struct Solution {}

impl Solution {
    pub fn num_to_digits(mut num: i32) -> Vec<i32> {
        let mut digits = Vec::new(); 
        while num > 0 {
            digits.push(num % 10);
            num /= 10; 
        }
        digits
    }
    pub fn add_digits(num: i32) -> i32 {
        let mut result = num; 
        while result >= 10 {
            let digits = Solution::num_to_digits(result);
            result = digits.iter().fold(0, |total, i| total + i); 
        }
        result 
    }
}

fn main() {
    assert_eq!(Solution::add_digits(38), 2); 
    assert_eq!(Solution::add_digits(10), 1);
}